_posts/tutorials/2017-11-17-javalin-docker-example.md

```diff
@@ -0,0 +1,173 @@
+                            </transformer>
+                        </transformers>
+                    </configuration>
+                </execution>
+            </executions>
+        </plugin>
+    </plugins>
+</build>
+~~~
+
+## Configuring Docker
+Before we can configure anything we must create a Dockerfile.
+we can create a text file using any editor and name it https://github.com/prasad-marne/javalin-docker/blob/master/Dockerfile 
+Copy below contents to the file and move this file to root of your project.
+~~~markup
+FROM openjdk:8-jre-alpine
+
+EXPOSE 7000
+ENTRYPOINT ["/usr/bin/java", "-jar", "/usr/share/javalin/my-javalin.jar"]
+
+ARG JAR_FILE
+ADD target/${JAR_FILE} /usr/share/javalin/my-javalin.jar
+~~~
+When you've added the Dockerfile to your project,
+it should look like [this](https://github.com/prasad-marne/javalin-docker/blob/master/Dockerfile)
+
+To build a docker image for your application we will use dockerfile-maven-plugin
+ [(dockerfile-maven-plugin)](https://github.com/spotify/dockerfile-maven).
+ Set DOCKER_HOST environment variable as mentioned [(here)](https://github.com/spotify/docker-maven-plugin#setup).
+ In the repository section "localhost:5000" is the registry url.
+~~~markup
+<plugin>
+    <groupId>com.spotify</groupId>
+    <artifactId>dockerfile-maven-plugin</artifactId>
+    <version>1.3.6</version>
+    <executions>
+        <execution>
+            <id>default</id>
+            <goals>
+                <goal>build</goal>
+                <goal>push</goal>
+            </goals>
+        </execution>
+    </executions>
+    <configuration>
+        <repository>localhost:5000/javalin</repository>
+        <tag>${project.version}</tag>
+        <buildArgs>
+            <JAR_FILE>${project.build.finalName}-shaded.jar</JAR_FILE>
+        </buildArgs>
+    </configuration>
+</plugin>
+~~~
+When you've added the Docker config to your pom,
+it should look like [this](https://github.com/prasad-marne/javalin-docker/blob/master/pom.xml)
+
+## Making Javalin Listen on the Correct Port
+The only thing left is making sure Javalin can handle your requests.
+We have exposed port 7000 in Dockerfile. That means that 7000 port on the container is accessible to the outside world.
+So we will configure Javalin to listen on "7000"
+~~~java
+import io.javalin.Javalin;
+
+public class HelloWorld {
+
+    public static void main(String[] args) {
+        Javalin app = Javalin.start(7000);
+        app.get("/", ctx -> ctx.result("Hello World"));
+    }
+
+}
+~~~
+## Build and Push Docker image
+Now we can deploy our application using `mvn deploy`.
+This will build the docker image and push it to your registry server.
+Image name is same as repository value in the pom. 
+Additionally we add a tag to image to specify images for different versions.
+So image name for this example is localhost:5000/javalin:1.0.0-SNAPSHOT.
+Again, make sure you're in your project root, then enter:
+~~~bash
+mvn deploy
```

Comment:
When doing this, I get

> [ERROR] Failed to execute goal com.spotify:dockerfile-maven-plugin:1.3.6:build (default) on project javalin-docker-example: Could not build image: java.util.concurrent.ExecutionException: com.spotify.docker.client.shaded.javax.ws.rs.ProcessingException: org.apache.http.conn.HttpHostConnectException: Connect to localhost:2375 [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused: connect -> [Help 1]

Reply:
can you share your output for of this command "docker ps". Also have you set DOCKER_HOST env variable?